src/v/resource_mgmt/rate.h

```diff
@@ -41,12 +41,15 @@ class rate_tracker final {
       , _window_size(window_size)
       , _current(0) {}
 
-    // record an observation and return the updated rate in units/second.
-    double record_and_measure(double v, const clock::time_point& now) {
+    // record an observation.
+    void record(double v, const clock::time_point& now) {
         // update the current window
         maybe_advance_current(now);
         _windows[_current].count += v;
+    }
 
+    // return the updated rate in units/second.
+    double measure(const clock::time_point& now) {
```

Comment:
`const`?

Reply:
Cannot be const because it purges old windows